Production_scripts/Social_Care/Care_Home.R

```diff
@@ -0,0 +1,299 @@
+#####################################################
+# Care Home Data
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+# Load packages
+library(dplyr)
+library(dbplyr)
+library(createslf)
+library(lubridate)
+library(tidyr)
+library(phsmethods)
+
+
+latest_update <- "Mar_2022"
+
+
+# Read in data---------------------------------------
+
+# set-up conection to platform
+db_connection <- phs_db_connection(dsn = "DVPROD")
+
+# read in data - social care 2 care home
+ch_data <- tbl(db_connection, in_schema("social_care_2", "carehome_snapshot")) %>%
+  select(
+    ch_name,
+    ch_postcode,
+    sending_location,
+    social_care_id,
+    financial_year,
+    financial_quarter,
+    period,
+    ch_provider,
+    reason_for_admission,
+    type_of_admission,
+    nursing_care_provision,
+    ch_admission_date,
+    ch_discharge_date,
+    age
+  ) %>%
+  # correct FY 2017
+  mutate(financial_quarter = if_else(financial_year == 2017 & is.na(financial_quarter), 4, financial_quarter)) %>%
+  mutate(period = if_else(financial_year == 2017 & financial_quarter == 4, "2017Q4", period)) %>%
+  # filter missing admission dates
+  filter(!is.na(ch_admission_date)) %>%
+  collect()
+
+
+# Data Cleaning ---------------------------------------
+
+# period start and end dates
+period_dates <- ch_data %>%
+  distinct(period) %>%
+  mutate(
+    record_date = yq(period) %m+% period(6, "months") %m-% days(1),
+    qtr_start = yq(period) %m+% period(3, "months")
+  )
+
+ch_clean <- ch_data %>%
+  left_join(period_dates, by = c("period")) %>%
+  # filter out any episodes where discharge is before admission
+  mutate(dis_before_adm = ch_admission_date > ch_discharge_date & !is.na(ch_discharge_date)) %>%
+  filter(dis_before_adm == FALSE) %>%
+  # sort data
+  arrange(sending_location, social_care_id, period, ch_admission_date, ch_discharge_date)
+
+
+# Match with demographics data  ---------------------------------------
+
+# read in demographic data
+sc_demog <- haven::read_sav(get_sc_demog_lookup_path()) # remaining here until .rds file ready
+sc_demog <- readr::read_rds(get_sc_demog_lookup_path())
+
+matched_ch_data <- ch_clean %>%
+  left_join(sc_demog, by = c("sending_location", "social_care_id"))
+
+
+# Data Cleaning Care Home Name and Postcode ---------------------------------------
+
+matched_ch_clean <- matched_ch_data %>%
+  # correct postcode formatting
+  mutate(across(contains("postcode"), .x = format_postcode(.x)))
```

Comment:
```suggestion
matched_ch_data <- ch_clean %>%
  left_join(sc_demog, by = c("sending_location", "social_care_id"))%>%
  # correct postcode formatting
  mutate(across(contains("postcode"), .x = format_postcode(.x)))
```
I think this line can be added into the previous section to cut down the number of objects. I was getting a bit confused with which dataframe i was looking at